### DISCUSSION TITLE
header style with row view

### DISCUSSION THREAD

**Discussion by sotten** [rappasoft/laravel-livewire-tables#340] [Q&A]
is it possible to change the style of the header in a row view?
For the cells it is possible to pass a property.
```
<x-livewire-tables::table.cell :customAttributes="['class' => 'my-classes']">
    {{ $row->id }}
</x-livewire-tables::table.cell>
```
Is this also possible for the header without modifying the `heading.blade.php` file?

**Comment by rappasoft** [comment]
You can try addClass: https://github.com/rappasoft/laravel-livewire-tables/wiki/Misc.-Column-Functionality

**Reply by sotten** [reply]
thank you for the quick reply.

I have the following situation:
- I use several tables, on different pages.
- One of these tables has many columns >20, which does not fit on the screen without scrolling.
- I can adjust the data rows formatting (reduce margin and font size).
- The tables heading can also be adjusted after publishing `heading.blade.php`. But this change affect all tables, but I want to change only this specific table heading.

I mainly want to reduce the margin, font size and tracking to make the column width smaller.
<br>

This
https://github.com/rappasoft/laravel-livewire-tables/blob/a840e419bd6bcac33cde9ca1b67a92b0a29590bf/resources/views/tailwind/components/table/heading.blade.php#L12
and this one,
https://github.com/rappasoft/laravel-livewire-tables/blob/a840e419bd6bcac33cde9ca1b67a92b0a29590bf/resources/views/tailwind/components/table/heading.blade.php#L19
I want to customize for a single table.

<br>
With addClass, I can only customize the following:

https://github.com/rappasoft/laravel-livewire-tables/blob/a840e419bd6bcac33cde9ca1b67a92b0a29590bf/resources/views/tailwind/components/table/heading.blade.php#L9

I hope I could describe my problem better now. Do you have a suggested solution, or did I misunderstand your answer?

**Reply by rappasoft** [reply]
You can try to override the margin and padding using the addClass of each column for the specific tables. If that doesn't help you might have to publish the view for now. Or just add global styles to the page of the tables you want targeting `th` and change them from there.

**Reply by fefo-p** [reply]
How about in v2 as addClass has been removed?

**Reply by inmanturbo** [reply]
https://rappasoft.com/docs/laravel-livewire-tables/v2/datatable/available-methods#content-settheadattributes

```
public function configure(): void
{
  $this->setTheadAttributes([
    'id' => 'my-id',
    'class' => 'this that',
  ]);
}
```

**Reply by fefo-p** [reply]
I guess this can be applied to only one column as well then.
I'll try.

Thanks!
